### PR TITLE
Rebuilding the Dashboard

### PR DESCRIPTION
**Pull request by Ahmed-Elhalwagy**

### CODE REVIEW

**Review by Mahmoudgalalz — CHANGES_REQUESTED** [review]
The table is shaking, with no resone, also non of the functions works at all, and cannot delete, update 

and I cannot confirm, there's no selection

let me know if you are working on these

There's types error

**Comment by Ahmed-Elhalwagy** [comment]
I am working on these existing problems

File: components/DataTable.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+import useSWR,{mutate} from 'swr'
+import {fetcher} from '@/lib/utils'
+import TableButtons from './TableButtons';
+import Swal from 'sweetalert2'
+import DeleteIcon from '@mui/icons-material/Delete';
+import SaveIcon from '@mui/icons-material/Save';
+
+
+const bulk:CrawledOpportunity[] = [{
+  id:'123123',
```
Comment (Mahmoudgalalz) [inline components/DataTable.tsx]: Remove the bulk data

File: components/DataTable.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+      toast.success('Job updated successfully')
+    }
+    else {
+      console.log('fail')
+      toast.error('Job update failed')
+    }
+};
+
+const handleSave = async (row: { id: string; })=> {
+  console.log('Save Done');
```
Comment (Mahmoudgalalz) [inline components/DataTable.tsx]: Waiting for the #49

File: components/DataTable.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+      { field: `title`, headerName: 'Title', width: 280 , editable: true },
+      { field: `link`, headerName: 'Link', width: 250, editable: true,
+        renderCell: (params: GridRenderCellParams) => (
+          <a 
+            href={params.row.link} 
+            target="_blank" 
+            rel="noopener noreferrer" 
+            style={{ textDecoration: 'none' }}
+            onMouseEnter={(e) => {
+              const target = e.target as HTMLElement;
```
Comment (Mahmoudgalalz) [inline components/DataTable.tsx:190]: Specify the type of the Element

File: components/TableButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+interface TableButtons {
+  selectedRows :GridRowId[]
+  setCrawlerJobs :Dispatch<SetStateAction<CrawledOpportunity[]>>
+  setSelectedRows: Dispatch<SetStateAction<GridRowId[]>>
+  crawlerJobs:CrawledOpportunity[]
+}
+function TableButtons({selectedRows, setCrawlerJobs,setSelectedRows, crawlerJobs}: TableButtons){
+
+  const newJob= {
```
Comment (Mahmoudgalalz) [inline components/TableButtons.tsx]: You add a plain job, I want to send data from the dashboard that I create

**Review by Mahmoudgalalz — CHANGES_REQUESTED** [review]
I asked for some changes, and there's change's
I wanted to fix the adding feature, I don't why you have implemented with this way, there's a way to add a row within the table then once it's new row and we click on the save button it will save, not update it

also we need a button to confirm the job, within the action buttons

**Pull request merged**